src/infrastructure/NodeService.js

```diff
@@ -0,0 +1,103 @@
+/*
+ *
+ * Copyright (c) 2019-present for NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License ");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import http from './http'
+import Constants from '../config/constants'
+import * as symbol from 'symbol-sdk'
+
+class NodeService {
+    /**
+     * Get Storage Info from symbol SDK
+     * @returns StorageInfo
+     */
+    static getStorageInfo = () => {
+      return http.node.getStorageInfo().toPromise()
+    }
+
+    /**
+     * Get Node Info from symbol SDK
+     * @returns NodeInfo
+     */
+    static getNodeInfo = () => {
+      return http.node.getNodeInfo().toPromise()
+    }
+
+    /**
+     * Get Server Info from symbol SDK
+     * @returns ServerInfo
+     */
+    static getServerInfo = () => {
+      return http.node.getServerInfo().toPromise()
+    }
+
+    /**
+     * Get Node Peers from symbol SDK
+     * @returns NodeInfo[]
+     */
+    static getNodePeers = async () => {
+      const nodePeers = await http.node.getNodePeers().toPromise()
+
+      const formattedNodePeers = nodePeers.map(nodeInfo => this.formatNodeInfo(nodeInfo))
+
+      return formattedNodePeers
+    }
+
+    /**
+     * Get node health status by endpoint.
+     * @param string api-node endpoint such as http:localhost:3000
+     * @returns boolean
+     */
+    static isNodeActive = async (currentUrl) => {
+      let status = true
+
+      try {
+        await new symbol.NodeHttp(currentUrl).getNodeHealth().toPromise()
```

Comment:
I've found out that some **working** nodes throw the 500 error as a response to this

Reply:
i see, create new issue for this.